### ISSUE TITLE
Provide grammar to Linguist project

### ISSUE THREAD

**Issue by Lordfirespeed** [zenstackhq/zenstack#823]
The [Linguist](https://github.com/github-linguist/linguist/) project is what GitHub uses for language detection and third-party grammar selection.

We can enable syntax highlighting for the ZenStack language on GitHub by providing a language grammar to this project.

**Comment by jiashengguo** [comment]
@Lordfirespeed, thanks for the info.  However, there is a bar for them to accept a new extension, which I think is still far for ZenStack:

https://github.com/github-linguist/linguist/issues/5756

**Comment by ymc9** [comment]
Maybe we're not too far from this rule?

> at least 2000 files per extension indexed in the last year (the number you see at the top of the search results), unless the extension is expected to only occur once per repo, then 200 files.

**Comment by Lordfirespeed** [comment]
I agree, the language isn't far off - although it's unclear whether the listed rules are OR or AND

**Comment by ymc9** [comment]
PR pending here: https://github.com/github-linguist/linguist/pull/7065
